src/App.tsx

```diff
@@ -1,26 +1,304 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { FormEvent, useEffect, useRef, useState } from 'react';
+import {
+  USER_ID,
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { TodoItem } from './components/TodoItem';
 
-const USER_ID = 0;
+type Status = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState<Status>('all');
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+    getTodos()
+      .then((response: React.SetStateAction<Todo[]>) => setTodos(response))
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (!errorMessage) {
+      return;
+    }
+
+    const timeout = setTimeout(() => setErrorMessage(''), 3000);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [errorMessage]);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  let filterTodos = todos;
```

Comment:
```suggestion
  let filteredTodos = todos;
```
filterTodos, with verb in name sounds like a function name